core/google/cloud/client.py

```diff
@@ -72,33 +71,32 @@ def from_service_account_json(cls, json_credentials_path, *args, **kwargs):
 class Client(_ClientFactoryMixin):
     """Client to bundle configuration needed for API requests.
 
-    Assumes that the associated ``_connection_class`` only accepts
-    ``http`` and ``credentials`` in its constructor.
+    Stores ``credentials`` and ``http`` object so that subclasses
+    can pass them along to a connection class.
 
-    :type credentials: :class:`google.auth.credentials.Credentials` or
-                       :class:`NoneType`
-    :param credentials: The OAuth2 Credentials to use for the connection
-                        owned by this client. If not passed (and if no ``http``
-                        object is passed), falls back to the default inferred
-                        from the environment.
+    :type credentials: :class:`~google.auth.credentials.Credentials`
+    :param credentials: (Optional) The OAuth2 Credentials to use for this
+                        client. If not passed (and if no ``http`` object is
+                        passed), falls back to the default inferred from the
+                        environment.
 
-    :type http: :class:`httplib2.Http` or class that defines ``request()``.
-    :param http: An optional HTTP object to make requests. If not passed, an
+    :type http: :class:`~httplib2.Http`
+    :param http: (Optional) HTTP object to make requests. Can be any object
+                 that defines ``request()`` with the same interface as
+                 :meth:`~httplib2.Http.request`. If not passed, an
                  ``http`` object is created that is bound to the
                  ``credentials`` for the current object.
     """
 
-    _connection_class = Connection
-
     def __init__(self, credentials=None, http=None):
         if (credentials is not None and
                 not isinstance(
                     credentials, google.auth.credentials.Credentials)):
             raise ValueError(_GOOGLE_AUTH_CREDENTIALS_HELP)
         if credentials is None and http is None:
             credentials = get_credentials()
-        self._connection = self._connection_class(
-            credentials=credentials, http=http)
+        self._credentials = credentials
+        self._http = http
```

Comment:
In the `Connection` base class: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/8e258d5d50c909d1125d58e44a82a86f148c4f91/core/google/cloud/_http.py#L96-L108

It's worth noting: this code does exactly what it did before the change, but now the subclasses call the `Connection` constructor instead of the base class (and also the base class stores two new attributes `_credentials` and `_http`)